api/types/swarm/swarm.go

```diff
@@ -153,6 +158,8 @@ type InitRequest struct {
 	Spec             Spec
 	AutoLockManagers bool
 	Availability     NodeAvailability
+	DefaultAddrPool  []*net.IPNet
```

Comment:
Although I suggested to use IPNet for as internal type (mainly for the CLI-side); for the over-the-wire format for this is rather awkward; the _string_ representation (without converting individual entries to a string when marshalling);

For example, the request now needs to look like;

```json
{"DefaultAddrPool":[{"IP":"192.0.2.0","Mask":"////AA=="}]}
```

Whereas individually encoding IPNet to a string would look like

```json
{"DefaultAddrPool":["192.0.2.0/24"]}
```

Reply:
Completely agree with @thaJeztah, we shouldn't leak implementation detail in our API and have our own type here (if we need something complex). Here it seems just a slice of string would work.